src/librustc_mir/build/expr/into.rs

```diff
@@ -141,15 +141,16 @@ impl<'a,'tcx> Builder<'a,'tcx> {
                 //
 
                 let loop_block = this.cfg.start_new_block();
-                let exit_block = this.cfg.start_new_block();
+                let mut exit_block = this.cfg.start_new_block();
 
                 // start the loop
                 this.cfg.terminate(block, Terminator::Goto { target: loop_block });
 
-                this.in_loop_scope(loop_block, exit_block, |this| {
+                let may_term = unpack!(exit_block = this.in_loop_scope(loop_block,
+                                                                       exit_block,
```

Comment:
sorry to keep adding nits, but it seems like `exit_block` never changes, it just flows around and around. Maybe we should just change the closure's return type to `()`, and then we can make `in_loop_scope` return `bool` and drop this awkward `unpack`. The code will also be clearer in that it's clear that `exit_block` never changes (and I can't see why you would want it to?)